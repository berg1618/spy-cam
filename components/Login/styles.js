import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
    b:{
        height: '100%',
        backgroundColor: '#413C45',
        margin: 'auto',
        alignItems:'center',
    },
    titleApp: {
        fontSize: 55,
        textAlign: 'center',
        color:'#F2F0F4',
        marginTop:100,
        fontWeight:'800',
    },
    logo:{
        marginTop:48,
        marginBottom:62,
        width: 70,
        height:70,
    },
    formContainer: {
        width:'80%',
        backgroundColor:'#F2F0F4',
        position: 'relative',
        display: 'flex',
        alignItems: 'center',
        borderRadius: 10,
        padding:15,

    },
    titleContainer:{
        fontSize:20,
        color:'#413C45',
        fontWeight:'500',
        marginTop: 10,
        marginBottom: 20,
    },
    icon: {
        left: 2,
        top: 5,
        position: 'relative',
    },
    inputText: {
        marginLeft: 8,
        position:'relative',
        width: 150,
    },
    entrada: {
        flexDirection:'row',
        marginBottom: -12,
        marginTop: 10,
    },
    botao: {
        textAlign:'center',
        backgroundColor:'#413C45',
        color:'#F2F0F4',
        borderRadius: 10,
        paddingTop: 6,
        paddingBottom: 6,
        paddingLeft: 45,
        paddingRight: 45,
        marginTop:45,
        marginBottom: 5,
    },
    mensagem: {
        fontSize: 10,
    },
    textBold:{
        color:'#0057FF',
        fontSize: 9,
    },
})

export default styles