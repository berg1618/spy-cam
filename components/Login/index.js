import { Image, Text, View, TextInput, TouchableOpacity } from "react-native";
import styles from './styles'
import React from "react";

const Login = () => {
  return (
    <View style={styles.b}>
      <View>
        <Text style={styles.titleApp}>
          Spy Cam
        </Text>
      </View>

      <Image 
        style={styles.logo}
        source={require('../../assets/iconReconhecimento.png')}
        />

        <View style={styles.formContainer}>
          
          <Text style={styles.titleContainer}>Faça seu Login</Text>
          <View style={styles.entrada}>
            <Image
              style={styles.icon}
              source={require('../../assets/iconNome.png')}
            />
            <TextInput
              style={styles.inputText}
              placeholder="Nome"
              keyboardType="text"
            />
          </View>
          <View> 
          <Text>____________________________</Text>
          </View>

         
          <View style={styles.entrada}>
            <Image
              style={styles.icon}
              source={require('../../assets/iconPassword.png')}
            />
            <TextInput
              style={styles.inputText}
              placeholder="Senha"
              keyboardType="text"
            />
          </View>
          <View> 
          <Text>____________________________</Text>
          </View>

          <TouchableOpacity>
              <Text style={styles.botao}>Login</Text>
          </TouchableOpacity>

          <Text style={styles.mensagem}>
            Não possui conta? <Text style={styles.textBold}>Cadastre-se</Text>
          </Text>
         
        </View>
    </View>
  );
}

export default Login