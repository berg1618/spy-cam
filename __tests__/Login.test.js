import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../components/Login';


test('testando o componente de login', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });