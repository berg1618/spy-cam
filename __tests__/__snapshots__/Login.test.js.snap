// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testando o componente de login 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#413C45",
      "height": "100%",
      "margin": "auto",
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#F2F0F4",
          "fontSize": 55,
          "fontWeight": "800",
          "marginTop": 100,
          "textAlign": "center",
        }
      }
    >
      Spy Cam
    </Text>
  </View>
  <Image
    source={
      {
        "testUri": "../../../assets/iconReconhecimento.png",
      }
    }
    style={
      {
        "height": 70,
        "marginBottom": 62,
        "marginTop": 48,
        "width": 70,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F2F0F4",
        "borderRadius": 10,
        "display": "flex",
        "padding": 15,
        "position": "relative",
        "width": "80%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#413C45",
          "fontSize": 20,
          "fontWeight": "500",
          "marginBottom": 20,
          "marginTop": 10,
        }
      }
    >
      Faça seu Login
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": -12,
          "marginTop": 10,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/iconNome.png",
          }
        }
        style={
          {
            "left": 2,
            "position": "relative",
            "top": 5,
          }
        }
      />
      <TextInput
        keyboardType="default"
        placeholder="Nome"
        style={
          {
            "marginLeft": 8,
            "position": "relative",
            "width": 150,
          }
        }
      />
    </View>
    <View>
      <Text>
        ____________________________
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": -12,
          "marginTop": 10,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/iconPassword.png",
          }
        }
        style={
          {
            "left": 2,
            "position": "relative",
            "top": 5,
          }
        }
      />
      <TextInput
        keyboardType="default"
        placeholder="Senha"
        style={
          {
            "marginLeft": 8,
            "position": "relative",
            "width": 150,
          }
        }
      />
    </View>
    <View>
      <Text>
        ____________________________
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#413C45",
          "borderRadius": 10,
          "color": "#F2F0F4",
          "marginBottom": 5,
          "marginTop": 45,
          "opacity": 1,
          "paddingBottom": 6,
          "paddingLeft": 45,
          "paddingRight": 45,
          "paddingTop": 6,
          "textAlign": "center",
        }
      }
    >
      <Text>
        Login
      </Text>
    </View>
    <Text
      onPress={[Function]}
      style={
        {
          "fontSize": 10,
        }
      }
    >
      Não possui conta? 
      <Text
        style={
          {
            "color": "#0057FF",
            "fontSize": 9,
          }
        }
      >
        Cadastre-se
      </Text>
    </Text>
  </View>
</View>
`;
